src/libraries/System.Private.CoreLib/src/System/Threading/Thread.cs

```diff
@@ -72,6 +88,13 @@ private void RunWorker()
 
                 try
                 {
+#if TARGET_OSX
```

Comment:
```suggestion
#if TARGET_OSX || NATIVEAOT
```
We should enable this path for NAOT too since it is limited to setting the name of the current thread accross all OSes 
https://github.com/dotnet/runtime/blob/bf7d30b9ecfbb502f055fef74190754e85ff7bda/src/coreclr/nativeaot/System.Private.CoreLib/src/System/Threading/Thread.NativeAot.cs#L202